Sources/Apodini/SyntaxTreeVisitor/SyntaxTreeVisitor.swift

```diff
@@ -67,4 +81,26 @@ class SyntaxTreeVisitor {
             fatalError("Tried exiting a ContextNode which didn't have any parent nodes")
         }
     }
+    
+    
+    func formHandlerIndexPathForCurrentNode() -> HandlerIndexPath {
+        let rawValue = currentNodeIndexPath
```

Comment:
`reduce` + `dropLast`?

Reply:
what do you mean?

Reply:
I was thinking for something like: `var rawValue = currentNodeIndexPath.reduce("") { "\($0)\(max($1, 1) - 1)):" }` and then dropping the last character. You could spare iterating three times over the array. And after that, either marking the function as private since you use it only in STV or directly do the construction inside `register(handler:)`

Reply:
You're right that there's unnecessary iterations over the array (I got it down to 2 now, plus the string concatenation).

Not a fan of reduce though, at least in this specific case. I was trying not to be too clever w/ the code and i think piping it through the `map` and `joined(separator:)` calls is easier to read than using a `reduce`-based approach.

(Also, not to be too pedantic, but I think that, from a technical point of view, your solution, while requiring only 1 iteration over the array, would require a lot more expensive string constructions (since it always copies the result into the next result), and then also a full iteration over the String's characters, as well as another string concatenation at the end (String.dropLast returns an `[Character]`, which would need to be reassembled into a String). I could be wrong though. Also, since we probably won't have components nested more than a few dozen levels deep at most, i doubt the difference would ever be noticable.)